### ISSUE TITLE
Sharing token between extensions

### ISSUE THREAD

**Issue by fcollonval** [jupyterlab/lumino#168]
## Description

I got a [question](https://github.com/jupyterlab-contrib/jupyterlab-tour/issues/13) from an user wanting to import the token from the `jupyterlab-tour` extension because it was failing with the error: 

```
Error: No provider for: jupyterlab-tour:ITourManager.
```

I created a tiny example to reproduce the problem using:

```
cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts
mamba create -n wtour jupyterlab=3 jupyter-packaging nodejs jupyterlab-tour
conda activate wtour
cd with_tour
jlpm add jupyterlab-tour
vi src/index.ts  # Modify the code to require the extension token
jlpm run build
jupyter labextension develop . --overwrite
```

Loading JupyterLab, I hit the `No provider ...` error too.

Digging into Lumino Application code, the token content matches the one of the `_serviceMap` but the `get` returned `undefined`:

![image](https://user-images.githubusercontent.com/8435071/114889094-1854bc00-9e0a-11eb-951e-fd1e9c3aa134.png)

I guess the equality test is against the object pointer and not the object content. But I cannot figure out what is wrong.

## Reproduce

A minimal extension is available there: https://github.com/fcollonval/with_tour

## Expected behavior

The `with_tour` extension can require the `jupyterlab-tour:ITourManager` token.

## Context

- Operating System and version: Debian 10 buster
- Browser and version: Firefox 87.0
- JupyterLab version: 3.0.14

<details>
<summary>Web Console Output
</summary>
Error: No provider for: jupyterlab-tour:ITourManager.
    at W.e.resolveRequiredService (static/lab/jlab_core.20dcbc2dd05d71e94169.js?v=20dcbc2dd05d71e94169:2)
    at static/lab/jlab_core.20dcbc2dd05d71e94169.js?v=20dcbc2dd05d71e94169:2
    at Array.map (<anonymous>)
    at W.e.activatePlugin (static/lab/jlab_core.20dcbc2dd05d71e94169.js?v=20dcbc2dd05d71e94169:2)
    at static/lab/jlab_core.20dcbc2dd05d71e94169.js?v=20dcbc2dd05d71e94169:2
    at Array.map (<anonymous>)
</details>

**Comment by jasongrout** [comment]
> I guess the equality test is against the object pointer and not the object content.

That is correct. The common problem here is that there are two copies of the same package, one of which is activated by jlab (and thus supplies a token), and the other is used by the requesting package. Often this happens if the version numbers are not aligned, so webpack includes multiple copies of the package. A quick glance at your packages seem to indicate all is in order with the version numbers. To debug this further, I'd try searching your webpacked files to see where an extra copy of jupyterlab-tour might be coming from.

**Comment by jasongrout** [comment]
In this case, however, change this line: 

https://github.com/fcollonval/with_tour/blob/b8c5dc8fc1ec62a18d6b094645f2fcbc3b70e0c4/src/index.ts#L6

from 
```js
import { ITourManager } from 'jupyterlab-tour/lib/tokens';
```
to
```js
import { ITourManager } from 'jupyterlab-tour';
```
This is a known issue with prebuilt extensions in jlab. See https://github.com/jupyterlab/jupyterlab/pull/9475. Basically, you need to import from the top level, not from a file inside a package.

**Comment by jasongrout** [comment]
It looks like jupyterlab-tour will also need to export the token from the top level.

**Comment by fcollonval** [comment]
Thanks a lot @jasongrout for the explanation and the fix.

**Comment by jasongrout** [comment]
(If that fix doesn't work, let's open an issue in jlab, since this is specific to jlab not deduplicating files when you import them from inside a module)

**Comment by fcollonval** [comment]
Thanks again @jasongrout I published jupyterlab-tour@3.0.1 with the fix and it works.
